server/jetstream_cluster.go

```diff
@@ -5996,11 +5996,11 @@ func (s *Server) jsClusteredStreamUpdateRequest(ci *ClientInfo, acc *Account, su
 		// Need to remap any consumers.
 		for _, ca := range osa.consumers {
 			// Ephemerals are R=1, so only auto-remap durables, or R>1, unless stream is interest or workqueue policy.
-			numPeers := len(ca.Group.Peers)
-			if ca.Config.Durable != _EMPTY_ || numPeers > 1 || cfg.Retention != LimitsPolicy {
+			replicas := ca.Config.replicas(cfg)
```

Comment:
Why the switch here?

Reply:
It seemed clearer and less error-prone to check the intended number of replicas from the assignments rather than the Raft state. I expect the overall effect to be about the same but might be more robust during a Raft recovery.

Reply:
The real number of peers is best represented by len(ca.Group.Peers).

Reply:
I might suggest reverting this one unless there is a real bug that we have identified.

Reply:
OK reverted!